docstr_coverage/configFile.py

```diff
@@ -0,0 +1,43 @@
+"""
+This module is for conf file .docstr.yaml
+"""
+import os
+
+import yaml
+
+
+def readConfigFile(paths, kwargs):
+    """
+    Read config file and assign all variable to paths, kwargs.
+
+    :param tuple paths: paths list
+    :param dict kwargs: options
+
+    :return (paths, kwargs)
+    :rtype (tuple, dict)
+    """
+    # if .docstr.yaml not present return initial paths, and kwargs
+    if not os.path.exists(".docstr.yaml"):
+        return paths, kwargs
+
+    with open(".docstr.yaml", "r") as stream:
+        try:
+            config = yaml.safe_load(stream)
+            for item in config.keys():
```

Comment:
If I read this correctly, potentially passed CLI args may be overridden by configuration in the configuration file.

In my experience, the typical priority order (from low to high) is as follows:

**Library Defaults** (Hardcoded) <  **Project Defaults** (Config File) < **Runtime Settings** (CLI Args) 

See for example the first sentence in the accepted answer in [here](https://stackoverflow.com/a/2322649/2576703). There's a number of use-cases I can think of to support this. Feel free to ask back / discuss if you do not agree. I'm happy to elaborate more.

Reply:
I agree. We should give CLI args priority over config files. I think this may be fixed by changing line 38 to 
```if item not in kwargs.keys()```, although there may be other changes that are necessary.

Reply:
I **guess** `if item not in kwargs.keys()` won't work because of clicks default values which are probably set and indistinguishable from user inputs at this stage? I'm not sure though - don't really know click and did not yet run the code. But if I am right, changing this is probably not going to be easy...

Reply:
This looks promising, but then again, I really do not know much more about click than it's name:

[Stackoverflow: Handling Config Files in Click](https://stackoverflow.com/a/46391887/2576703)

Reply:
I didn't find a good solution too, an other way to do that it's to create a dict with key = option name and value = default_value of option but its not a clean solution.

Reply:
Looks like you were both right. This was a bit trickier than I expected. It's just about working, so once I get everything sorted out I'll update you and we can decide how to tie everything together

Reply:
@mBouamama, when you get a chance, can you please bring this branch into line with master to pull in #43? I know it seems unrelated, but it looks like we need it for Click to accept new defaults from parsed yaml files. Still cleaning up our solution

Reply:
Alright, @mBouamama and @MiWeiss, here's a seemingly working solution. I didn't commit these changes to @mBouamama's fork because I wanted you two to decide if you like it first. However, if it would be easier for you to review, just let me know and I can commit the changes outlined below: 

<details>
   <summary>Click me</summary>
   
   1. Add changes from PR #43
   2. In `cli.py`, change import on line 11 to `from docstr_coverage.configFile import set_config_defaults`
   3. In `cli.py`, make lines 161-177 look like this (with existing code for reference):
      ```python
      @click.help_option("-h", "--help")
      @click.argument(
          "paths",
          type=click.Path(exists=True, file_okay=True, dir_okay=True, readable=True, resolve_path=True),
          nargs=-1,
          # TODO: NEW CODE BELOW !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
          is_eager=True,  # Eagerly execute before `config` so `set_config_defaults` has `paths`
      )
      @click.option(
          "-C",
          "--config",
          type=click.Path(exists=False, resolve_path=True),
          default=".docstr.yaml",
          help="Configuration file containing option defaults",
          is_eager=True,
          callback=set_config_defaults,
      )
      # TODO: NEW CODE ABOVE !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
      @click.option("--skipmagic", "skip_magic_old", is_flag=True, help="Deprecated. Use --skip-magic")
      ```
   4. Add the following to `configFile.py`:
         ```python
      def set_config_defaults(ctx, param, value):
          """Update CLI option defaults in `ctx` to config file values
      
          Parameters
          ----------
          ctx: click.Context
              Click Context object
          param: click.Parameter
              Click Parameter object (assumed to be `config`)
          value: String
              Path to the configuration file
      
          Returns
          -------
          String
              Path to the configuration file"""
          if (value is not None) and os.path.exists(value):
              with open(value) as f:
                  config_data = yaml.safe_load(f) or {}
      
              try:
                  # Check if `paths` were given in config file
                  config_paths = config_data.pop("paths")
              except KeyError:
                  pass
              else:
                  # Use config default if `paths` was not provided as CLI argument
                  if (not ctx.params.get("paths")) and config_paths:
                      if isinstance(config_paths, str):
                          config_paths = [config_paths]
      
                      # Resolve paths like Click would have with the `click.Path.resolve_path` kwarg
                      ctx.params["paths"] = tuple([os.path.realpath(path) for path in config_paths])
              finally:
                  ctx.default_map = config_data
      
          return value
      ```
</details>

The `set_config_defaults` function basically just copies @mBouamama's `readConfigFile` and makes it work with Click. Some things that still need to be done:
1. Add some error handling to `set_config_defaults` like `readConfigFile`'s
2. Update `test_configFile.py`, since it stills tests the original `readConfigFile` function
3. Handle both ".yaml" and ".yml" extensions, since both are used

Like we all said, this turned out to be a bit more complicated than I (foolishly) initially thought. Let me know what you two think. If this works for you, @mBouamama can commit these changes with any updates, or I can commit them to @mBouamama's fork. Either way, @mBouamama already did all of the hard work, so you deserve all the credit. Sorry for the wall of text!

Reply:
Brilliant, @HunterMcGushion. Fixing it while @mBouamama and I are still chatting about a fix 🥇